airflow/providers/tableau/operators/tableau.py

```diff
@@ -0,0 +1,144 @@
+}
+
+
+class TableauOperator(BaseOperator):
+    """
+    Exectues a Tableau API Resource
+
+    .. seealso:: https://tableau.github.io/server-client-python/docs/api-ref
+
+    :param resource: The name of the resource to use.
+    :type resource: str
+    :param method: The name of the resource's method to execute.
+    :type method: str
+    :param resource_id: The id of resource wich will recive the action.
+    :type resource_id: str
+    :param resource_id_by: The resource field to be used as an id.
+    :type resource_id_by: Optional[str]
+    :param site_id: The id of the site where the workbook belongs to.
+    :type site_id: Optional[str]
+    :param blocking_refresh: By default the extract refresh will be blocking means it will wait until it has finished.
+    :type blocking_refresh: bool
+    :param tableau_conn_id: The :ref:`Tableau Connection id <howto/connection:tableau>`
+        containing the credentials to authenticate to the Tableau Server.
+    :type tableau_conn_id: str
+    """
+
+    def __init__(
+        self,
+        *,
+        resource: str,
+        method: str,
+        resource_id: str,
+        resource_id_by: str = 'id',
+        site_id: Optional[str] = None,
+        blocking_refresh: bool = True,
+        tableau_conn_id: str = 'tableau_default',
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.resource = resource
+        self.method = method
+        self.resource_id = resource_id
+        self.resource_id_by = resource_id_by
+        self.site_id = site_id
+        self.blocking_refresh = blocking_refresh
+        self.tableau_conn_id = tableau_conn_id
+
+    def execute(self, context: dict) -> str:
+        """
+        Executes the Tableau API resource and pushes the job id or downloaded file URI to xcom.
+        :param context: The task context during execution.
+        :type context: dict
+        :return: the id of the job that executes the extract refresh or downloaded file URI.
+        :rtype: str
+        """
+
+        available_resources = RESOURCES_METHODS.keys()
+        if self.resource not in available_resources:
+            error_message = f'Resource not found! Available Resources: {available_resources}'
+            raise AirflowException(error_message)
+
+        available_methods = RESOURCES_METHODS[self.resource]
+        if self.method in available_methods:
+            error_message = f'Method not found! Available methods for {self.resource}: {available_methods}'
+            raise AirflowException(error_message)
+
+        with TableauHook(self.site_id, self.tableau_conn_id) as tableau_hook:
+
+            resource = getattr(tableau_hook.server, self.resource)
+            method = getattr(resource, self.resource)
+
+            resource_id = self._get_resource_id_by(tableau_hook)
+
+            response = method(resource_id)
+
+        if self.method == 'download':
+            return response
```

Comment:
If I understand it right this means that you push to xcom the downloaded data source. I'm not tableau expert but isn't it may be massive amount of data? Should we apply limitation to it like `MAX_XCOM_SIZE` ? example in `GCSToLocalFilesystemOperator`: 

https://github.com/apache/airflow/blob/866a601b76e219b3c043e1dbbc8fb22300866351/airflow/providers/google/cloud/transfers/gcs_to_local.py#L134-L140

(Also another thought maybe we don't need to return anything when `method == 'download' `?)
WDYT?

Reply:
Whe using the download method, it push the file URI to xcoms. Usually is `/temp/the-tableau-file`.
The URI is not a huge string and doing that it enables you to use S3Operstor to backup the downloaded file

Reply:
OK got it. But how is it helpful to a downstream task? Tasks may run on diffrent workers thus having diffrent disks.

Reply:
Yeah, this is a problem. I'll remove this method for now